crates/turborepo-lib/src/cli/mod.rs

```diff
@@ -75,6 +75,35 @@ impl Display for OutputLogsMode {
     }
 }
 
+/// Which binary was used for the current invocation
+pub enum BinaryType {
+    /// Local binary called directly
+    Local,
+    /// Local binary was called via a global binary
+    LocalDefer,
+    /// Global binary called directly (No local binary to handoff to)
+    Global,
+    /// Global binary called because repo inference failed
+    GlobalFallback,
+    /// Repo inference was explicitly skipped via `--skip-infer`` arg
+    SkipInfer,
```

Comment:
I believe this is also capturing the case of `Local` when we detect that the local version supports `--skip-infer`. 

If we end up realizing in global turbo that we need to pass of to a local 1.7+, then we'll add a `--skip-infer` arg to the local call. [source](https://github.com/vercel/turbo/blob/main/crates/turborepo-lib/src/shim.rs#L640)

We check if that flag is set before we go into `run_correct_turbo` and [jump right in](https://github.com/vercel/turbo/blob/main/crates/turborepo-lib/src/shim.rs#L752).

I don't think `Local` will ever get populated since we didn't have telem in <1.7.